### PR TITLE
Sleep with Duration

### PR DESCRIPTION
**Pull request by pxp9**

### CODE REVIEW

File: src/lib.rs [hunk 1]
```diff
@@ -17,6 +17,7 @@ pub struct SleepParams {
     pub min_sleep_period: u64,
```
Comment (pxp9) [inline src/lib.rs:17]: Maybe this fields can be Duration ???

**Review by pxp9 — COMMENTED** [review]
Discuss Duration in SleepParams struct

File: src/lib.rs (hunk 1 above)
Comment (ayrat555) [reply src/lib.rs:17]: it's a great idea to use Duration. what do you think about using duration for all fields of the struct

Comment (pxp9) [reply src/lib.rs]: [SleepParams Duration :D](https://github.com/ayrat555/fang/pull/67/commits/c82180cc1440a7833ee93c45b5f4a7aac6115b19)

File: src/lib.rs [hunk 2]
```diff
@@ -12,11 +15,12 @@ impl Default for RetentionMode {
 }
 #[derive(Clone, Debug)]
 pub struct SleepParams {
-    pub sleep_period: u64,
-    pub max_sleep_period: u64,
-    pub min_sleep_period: u64,
-    pub sleep_step: u64,
+    pub sleep_period: Duration,
```
Comment (ayrat555) [inline src/lib.rs:18]: looks great. now readme should be updated

Comment (pxp9) [reply src/lib.rs:18]: [Readme updated!](https://github.com/ayrat555/fang/pull/67/commits/8a0c4b5cea557d28f204cb51028ad8aa9bab0dc1)

**Review by pxp9 — COMMENTED** [review]
:D

File: Cargo.toml [hunk 3]
```diff
@@ -17,6 +17,7 @@ blocking = ["diesel", "diesel-derive-enum", "dotenv"]
 asynk = ["bb8-postgres",  "postgres-types", "tokio", "async-trait", "typed-builder", "async-recursion"]
 
 [dependencies]
+time = "0.1.44"
```
Comment (ayrat555) [inline Cargo.toml:20]: I think this crate is not used.

Comment (pxp9) [reply Cargo.toml:20]: It is used for one for AsyncQueueError

File: src/blocking/queue.rs [hunk 4]
```diff
@@ -198,18 +199,21 @@ impl Queue {
...
         connection: &PgConnection,
-        error_margin_seconds: i64,
+        error_margin_seconds: StdDuration,
     ) -> Option<Vec<PeriodicTask>> {
         let current_time = Self::current_time();
 
-        let low_limit = current_time - Duration::seconds(error_margin_seconds);
-        let high_limit = current_time + Duration::seconds(error_margin_seconds);
+        let low_limit = current_time - Duration::from_std(error_margin_seconds).ok()?;
+        let high_limit = current_time + Duration::from_std(error_margin_seconds).ok()?;
```
Comment (ayrat555) [inline src/blocking/queue.rs:216]: I don't think it's a good approach to handle errors like this

File: src/blocking/queue.rs [hunk 5]
```diff
@@ -198,18 +199,21 @@ impl Queue {
             .ok()
     }
 
-    pub fn fetch_periodic_tasks(&self, error_margin_seconds: i64) -> Option<Vec<PeriodicTask>> {
+    pub fn fetch_periodic_tasks(
+        &self,
+        error_margin_seconds: StdDuration,
```
Comment (ayrat555) [inline src/blocking/queue.rs:204]: I think the name is wrong. since it's duration

Comment (pxp9) [reply src/blocking/queue.rs:204]: I forgot to change it

Comment (pxp9) [reply src/blocking/queue.rs]: [Error management](https://github.com/ayrat555/fang/pull/67/commits/b3fb2ab6b20328a26099ca8cce2f9f79d38cfc84)

File: src/blocking/queue.rs (hunk 4 above)
Comment (pxp9) [reply src/blocking/queue.rs]: [Error management](https://github.com/ayrat555/fang/pull/67/commits/b3fb2ab6b20328a26099ca8cce2f9f79d38cfc84)

File: Cargo.toml (hunk 3 above)
Comment (pxp9) [reply Cargo.toml]: delete it in [Error management](https://github.com/ayrat555/fang/pull/67/commits/b3fb2ab6b20328a26099ca8cce2f9f79d38cfc84)

**Review by ayrat555 — APPROVED** [review]
🚀

**Pull request merged**